bundles/binding/org.openhab.binding.radius/src/main/java/org/openhab/binding/radius/internal/RadiusBinding.java

```diff
@@ -0,0 +1,202 @@
+			logger.warn("Could not determine RADIUS server address");
+		}
+
+		sharedSecret = (String) configuration.get("sharedSecret");
+
+		if (StringUtils.isNotBlank(sharedSecret)) {
+			setProperlyConfigured(true);
+		} else {
+			setProperlyConfigured(false);
+		}
+
+		logger.debug("server={}, secret={}, refresh={}", radiusServer, sharedSecret, refreshInterval);
+	}
+
+	/**
+	 * @{inheritDoc
+	 */
+	@Override
+	protected long getRefreshInterval() {
+		return refreshInterval;
+	}
+
+	/**
+	 * @{inheritDoc
+	 */
+	@Override
+	protected String getName() {
+		return "Radius Refresh Service";
+	}
+
+	/**
+	 * @{inheritDoc
+	 */
+	@Override
+	protected void execute() {
+		for (RadiusBindingProvider provider : providers) {
+			for (String itemName : provider.getInBindingItemNames()) {
+				logger.debug("Item '{}' is about to be refreshed", itemName);
+				RadiusPacket p = new RadiusPacket(provider.getRadiusType(itemName));
+				RadiusPacket pRet = null;
+				p.addAttribute(new RadiusAVP(RadiusAttribute.USER_NAME, provider.getUserName(itemName)));
+				p.encodePapPassword(provider.getPassword(itemName), sharedSecret);
+				try {
+					DatagramSocket clientSocket = new DatagramSocket();
+
+					byte[] receiveData = new byte[1024];
+					DatagramPacket sendPacket = new DatagramPacket(p.getRawPacket(), p.getRawPacketLength(),
+							radiusServer, 1812);
+					logger.trace("Sent: {}", p);
+					clientSocket.send(sendPacket);
+					DatagramPacket receivePacket = new DatagramPacket(receiveData, receiveData.length);
+					clientSocket.receive(receivePacket);
+					clientSocket.close();
+					pRet = new RadiusPacket(receivePacket.getData());
+					logger.trace("Received: {}", pRet);
+				} catch (IOException e) {
+					logger.warn("Error while sending packet to server");
+				}
+
+				if (pRet == null) {
+					logger.warn("Received empty or no packet");
+					return;
+				}
+
+				Class<? extends Item> itemType = provider.getItemType(itemName);
+				State state = null;
+				byte attr = provider.getRadiusAttribute(itemName);
+
+				if ((attr == 0) || pRet.hasAttribute(attr)) {
+					if (itemType.isAssignableFrom(StringItem.class)) {
+						if (attr == 0) {
+							state = UnDefType.UNDEF;
+						} else {
+							state = new StringType(pRet.getAttribute(attr).getValueString());
+						}
+					} else if (itemType.isAssignableFrom(NumberItem.class)) {
+						if (attr == 0) {
+							state = UnDefType.UNDEF;
+						} else {
+							state = new DecimalType(pRet.getAttribute(attr).getValueLong());
+						}
+					} else if (itemType.isAssignableFrom(SwitchItem.class)) {
+						if (p.getTypeCode() == RadiusTypeCode.ACCESS_REQUEST) {
+							if (pRet.getTypeCode() == RadiusTypeCode.ACCESS_ACCEPT) {
+								state = OnOffType.ON;
+							} else if (pRet.getTypeCode() == RadiusTypeCode.ACCESS_REJECT) {
+								state = OnOffType.OFF;
+							} else {
+								state = UnDefType.UNDEF;
+							}
+						}
+					}
+				} else {
+					state = UnDefType.UNDEF;
+				}
```

Comment:
refactor and move this if-else into its own method (for the sake of readability)